cnudie/util.py

```diff
@@ -414,6 +416,121 @@ def find_changed_component(
+            f'{to_component_name}&view=diff'
+        )
+        bom_diff_header = f'## <a href="{dashboard_view_diff}">BoM Diff</a>\n'
+    else:
+        bom_diff_header = '## BoM Diff\n'
+
+    added_components = [
+        f'\U00002795 {component.name} (v{component.version})'
+        for component in component_diff.cidentities_only_right
+        if component.name not in component_diff.names_version_changed
+    ]
+
+    removed_components = [
+        f'\U00002796 {component.name} (v{component.version})'
+        for component in component_diff.cidentities_only_left
+        if component.name not in component_diff.names_version_changed
+    ]
+
+    changed_components = [
+        f'\U00002699 {name}: {old_component.version} → {new_component.version}'
+        for name in component_diff.names_version_changed
+        for old_component, new_component in component_diff.cpairs_version_changed
+        if old_component.name == name
+    ]
+
+    summary_counts = (
+        f'Added components: {len(added_components)}\n'
+        f'Changed components: {len(changed_components)}\n'
+        f'Removed components: {len(removed_components)}\n\n'
+    )
+
+    summary_details = []
+    if added_components:
+        summary_details.append('### Added Components:\n' + '\n'.join(added_components) + '\n')
+
+    if removed_components:
+        summary_details.append('### Removed Components:\n' + '\n'.join(removed_components) + '\n')
+
+    if changed_components:
+        summary_details.append('### Changed Components:\n' + '\n'.join(changed_components) + '\n')
+
+    component_details = []
+    for old_component, new_component in component_diff.cpairs_version_changed:
+        component_name = f'{new_component.name}'
+        if delivery_dashboard_url:
+            component_link = (
+                f"<a href='{delivery_dashboard_url}/#/component?name={new_component.name}'>"
+                f'{component_name}</a>'
+            )
+        else:
+            component_link = component_name
+
+        component_header = (
+            f'<details><summary>\U00002699 {component_link}:'
+            f'{old_component.version} → {new_component.version}</summary>\n')
+
+        # Sort resources into categories: added, removed and changed
+        added_resources = []
+        removed_resources = []
+        changed_resources = []
+
+        for new_resource in new_component.resources:
+            matching_old_resource = next(
+                (old_resource for old_resource in old_component.resources
+                 if old_resource.name == new_resource.name), None
+            )
+
+            if matching_old_resource:
+                if matching_old_resource.version != new_resource.version:
+                    changed_resources.append(
+                        [f'\U0001F504 {new_resource.name}',
+                         f'{matching_old_resource.version} → {new_resource.version}'])
+            else:
+                added_resources.append(
```

Comment:
Also, this heuristics wouldn't determine if a resource was added in a new version, if the resource-name already existed in any version before, would it?